create_jenkins_job.py

```diff
@@ -159,15 +161,27 @@ def create_job(os_name, job_name, template_file, additional_dict):
         # configure manual triggered job
         create_job(os_name, 'ci_' + os_name, 'ci_job.xml.em', {
             'cmake_build_type': 'None',
+            'build_discard': {
+                'days_to_keep': 1000,
+                'num_to_keep': 3000,
+            },
         })
         # configure test jobs for experimenting with job config changes
         # Keep parameters the same as the manual triggered job above.
         create_job(os_name, 'test_ci_' + os_name, 'ci_job.xml.em', {
             'cmake_build_type': 'None',
+            'build_discard': {
+                'days_to_keep': 1000,
+                'num_to_keep': 3000,
+            },
         })
 
         # configure a manual version of the packaging job
         create_job(os_name, 'ci_packaging_' + os_name, 'packaging_job.xml.em', {
+            'build_discard': {
+                'days_to_keep': 1000,
+                'num_to_keep': 100,
```

Comment:
This is the only number which looks a bit short. How about keeping at least one year for now?

Reply:
Only the Linux packaging job is currently over 100 and builds and only the windows one is close to 100 jobs. If we want this to have any immediate usefulness it can't be raised.

Reply:
I didn't realize these are the `ci_packaging` jobs. They are only used to verify changes with a packaging build. We could even cut them down more: 180 days?

Reply:
No complaints from me for being stricter :+1: